test/e2e/list_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package e2e
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	uitest "github.com/cppforlife/go-cli-ui/ui/test"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAppListing(t *testing.T) {
+	env := BuildEnv(t)
+	logger := Logger{}
+	kapp := Kapp{t, env.Namespace, env.KappBinaryPath, logger}
+	yaml1 := `
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: redis-primary
+  labels:
+    x: "y"
+spec:
+  ports:
+  - port: 6380
+    targetPort: 6380
+  selector:
+    app: redis
+    tier: backend
+`
+	yaml2 := `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: redis-config
+  labels:
+    x: "z"
+data:
+  key: value
+`
+
+	name := "test-app-1"
+	name2 := "test-app-2"
+	cleanUp := func() {
+		kapp.Run([]string{"delete", "-a", name})
+		kapp.Run([]string{"delete", "-a", name2})
+	}
+
+	cleanUp()
+	defer cleanUp()
+	logger.Section("App listing and filter label", func() {
+		_, _ = kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name,
+			"--labels", "x=y"}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml1)})
+
+		_, _ = kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", name2,
+			"--labels", "a=b"}, RunOpts{IntoNs: true, StdinReader: strings.NewReader(yaml2)})
+
+		listedAppsWithAge, _ := kapp.RunWithOpts([]string{"ls", "--filter-age", "2m+", "--json"}, RunOpts{Interactive: true})
+
+		response := uitest.JSONUIFromBytes(t, []byte(listedAppsWithAge))
+
+		if len(response.Tables) > 0 {
+			require.Empty(t, response.Tables[0].Rows, "Expected table rows to empty")
+		}
+
+		time.Sleep(2 * time.Second)
```

Comment:
Done. was not giving consistent behaviour as second app is getting 5s+ older sometimes so checking for 10s+ to just get the first app (it can depend on the machine we are running on)